server/pulp/server/db/connection.py

```diff
@@ -84,13 +84,13 @@ def initialize(name=None, seeds=None, max_pool_size=None, replica_set=None, max_
         # attempt to authenticate to the database
         username = config.config.get('database', 'username')
         password = config.config.get('database', 'password')
-        if username and password:
+        if username:
             _logger.debug(_('Attempting username and password authentication.'))
             connection_kwargs['username'] = username
             connection_kwargs['password'] = password
-        elif (username and not password) or (password and not username):
+        elif (password and not username):
```

Comment:
I don't think this needs the () around it anymore. Previously that enforced an order of evaluation, but we can use default left to right evaluation here.